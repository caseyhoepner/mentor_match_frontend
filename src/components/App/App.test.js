import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './';

describe('App', () => {
  let wrapper;
  let mockThunk = jest.fn()
  let mockThunk2 = jest.fn()
  let mockThunk3 = jest.fn()
  let mockFunc4 = jest.fn()
  let mockSearch = { search: '?token=3' }
  
  beforeEach(() => {
    wrapper = shallow(<App 
      retrieveMentors={mockThunk} 
      retrieveStudents={mockThunk2} 
      retrieveRelationships={mockThunk3} 
      setToken={mockFunc4}
      location={mockSearch}
    />)
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps function', () => {
    it('should return an object with an array of mentorsand the mentor modalInfo', () => {
      const mockState = { 
        mentors: [ { name: 'Joan', preferences: { title: 'Being Joan' } } ],
        modalInfo: { name: 'Jo-jo', identity_preference: [ 'Veteran' ] },
        somethingElse: 'somethingElse' 
      }
      const expected = { 
        mentors: [ { name: 'Joan', preferences: { title: 'Being Joan' } } ],
        modalInfo: { name: 'Jo-jo', identity_preference: [ 'Veteran' ] }
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
  });

  describe('mapDispatchToProps function', () => {
    const mockDispatch = jest.fn()
    let mappedProps;

    beforeEach(() => {
      mappedProps = mapDispatchToProps(mockDispatch)
    });

    it('should call dispatch when retrieveMentors is called', () => {
      mappedProps.retrieveMentors()
      expect(mockDispatch).toHaveBeenCalled()
    });

    it('should call dispatch when retrieveStudents is called', () => {
      mappedProps.retrieveStudents()
      expect(mockDispatch).toHaveBeenCalled()
    });

    it('should call dispatch when retrieveRelationships is called', () => {
      mappedProps.retrieveRelationships()
      expect(mockDispatch).toHaveBeenCalled()
    });
  });
});