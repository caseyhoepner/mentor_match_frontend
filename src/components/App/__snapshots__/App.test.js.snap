// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    location={
      Object {
        "search": "?token=3",
      }
    }
    retrieveMentors={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    retrieveRelationships={
      [MockFunction] {
        "calls": Array [
          Array [
            "3",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    retrieveStudents={
      [MockFunction] {
        "calls": Array [
          Array [
            "3",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setToken={
      [MockFunction] {
        "calls": Array [
          Array [
            "3",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/new-mentor-form"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/new-student-form"
        />
        <Route
          path="/edit-mentor"
          render={[Function]}
        />
        <Route
          component={[Function]}
          path="/admin-dashboard"
        />
        <Route
          component={[Function]}
          path="/success"
        />
        <Route
          component={[Function]}
        />
      </Switch>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/new-mentor-form"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/new-student-form"
          />,
          <Route
            path="/edit-mentor"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            path="/admin-dashboard"
          />,
          <Route
            component={[Function]}
            path="/success"
          />,
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/new-mentor-form",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/new-student-form",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/edit-mentor",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin-dashboard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/success",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/new-mentor-form"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/new-student-form"
          />
          <Route
            path="/edit-mentor"
            render={[Function]}
          />
          <Route
            component={[Function]}
            path="/admin-dashboard"
          />
          <Route
            component={[Function]}
            path="/success"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/new-mentor-form"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/new-student-form"
            />,
            <Route
              path="/edit-mentor"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              path="/admin-dashboard"
            />,
            <Route
              component={[Function]}
              path="/success"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/new-mentor-form",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/new-student-form",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/edit-mentor",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/admin-dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/success",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
